File: Express/src/api/v1/models/Todo.js
const mongoose = require("mongoose");
const toJSON = require("./plugins/toJson");

const todoSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: [true, "Title is required"],
      trim: true,
    },
    description: {
      type: String,
      required: [true, "Description is required"],
      trim: true,
    },
    dueDate: {
      type: Date,
      default: null,
    },
    priority: {
      type: String,
      enum: ["Low", "Medium", "High"],
      default: "Medium",
    },
    status: {
      type: String,
      enum: ["Pending", "InProgress", "Completed"],
      default: "Pending",
    },
    tags: [
      {
        type: String,
        trim: true,
      },
    ],
    completed: {
      type: Boolean,
      default: false,
    },
    owner: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      ref: "User",
    },
  },
  { timestamps: true }
);

// Apply the toJSON plugin to the todo schema
todoSchema.plugin(toJSON);

const Todo = mongoose.model("Todo", todoSchema);

module.exports = Todo;

File: fastapi/fastapi-rest/app/schemas/todo.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
import enum

class PriorityEnum(str, enum.Enum):
    low = "Low"
    medium = "Medium"
    high = "High"

class StatusEnum(str, enum.Enum):
    pending = "Pending"
    in_progress = "InProgress"
    completed = "Completed"

class TodoBase(BaseModel):
    title: str
    description: str
    due_date: Optional[datetime] = None
    priority: PriorityEnum = PriorityEnum.medium
    status: StatusEnum = StatusEnum.pending
    tags: Optional[str] = None
    completed: bool = False

class TodoCreate(TodoBase):
    pass

class TodoUpdate(TodoBase):
    pass

class Todo(TodoBase):
    id: int
    owner_id: int

    class Config:
        from_attributes = True
